AWS/collectors/custom/ECS/Service.js

```diff
@@ -61,21 +61,28 @@ export const query = async (AWS_MAPPING, serviceName, resourceType, region) => {
             }
           });
         }
+        const taskDefs = [...new Set(taskDefinitionArns)];
+        updateResourceTypeCounter(
+          AWS_MAPPING,
+          serviceName,
+          ECS_TASK_DEFINITION,
+          { cspmUnits: taskDefs.length }
+        );
+        total += taskDefs.length;
 
         await Promise.all(
-          [...new Set(taskDefinitionArns)].map(async (taskDefinition) => {
+          taskDefs.map(async (taskDefinition) => {
             let getTaskDefCmd = new DescribeTaskDefinitionCommand({
               taskDefinition,
             });
             let getTaskDefRes = await client?.send(getTaskDefCmd);
             let containerCount =
               getTaskDefRes.taskDefinition?.containerDefinitions.length;
-            total += containerCount;
             updateResourceTypeCounter(
               AWS_MAPPING,
               serviceName,
               ECS_TASK_DEFINITION,
-              containerCount,
+              { cwppUnits: containerCount },
```

Comment:
just curious, shouldn't we do the same for ECS task-definitions? I mean counting task definitions instead of containers?

Reply:
Refer line 65 of this file.